docs/core/extensions/timer-service.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Implement the IHostedService interface
+description: Learn how to implement a custom IHostedService interface with .NET.
+author: IEvangelist
+ms.author: dapine
+ms.date: 05/26/2021
+ms.topic: tutorial
+---
+
+# Implement the `IHostedService` interface
+
+When you need finite control beyond the provided <xref:Microsoft.Extensions.Hosting.BackgroundService>, you can implement your own <xref:Microsoft.Extensions.Hosting.IHostedService>. The <xref:Microsoft.Extensions.Hosting.IHostedService> interface is the basis for all long running services in .NET. Custom implementations are registered with the <xref:Microsoft.Extensions.DependencyInjection.ServiceCollectionHostedServiceExtensions.AddHostedService%60%601(Microsoft.Extensions.DependencyInjection.IServiceCollection)> extension method.
+
+In this tutorial, you learn how to:
+
+> [!div class="checklist"]
+>
+> - Implement the <xref:Microsoft.Extensions.Hosting.IHostedService>, and <xref:System.IAsyncDisposable> interfaces.
+> - Create a timer-based service.
+> - Register the custom implemenation with dependency injection and logging.
+
+## Prerequisites
+
+- The [.NET 5.0 SDK or later](https://dotnet.microsoft.com/download/dotnet)
+- A .NET integrated development environment (IDE)
+  - Feel free to use [Visual Studio](https://visualstudio.microsoft.com)
+
+## Create timer service
+
+The timer-based background service makes use of the <xref:System.Threading.Timer?displayProperty=fullName> class. The timer triggers the `DoWork` method. The timer is disabled on <xref:Microsoft.Extensions.Hosting.IHostLifetime.StopAsync(System.Threading.CancellationToken)?displayProperty=nameWithType> and disposed when the service container is disposed on <xref:System.IAsyncDisposable.DisposeAsync?displayProperty=nameWithType>:
+
+Replace the contents of the `Worker` from the template with the following C# code, and rename the file to *TimerService.cs*:
```

Comment:
Add section on how to add the tempalte like scoped and queue service docs.

Reply:
Ah, yes! Great catch - will do